Bottom: a62ebf4f240ba1f0f9b8d8d71fbdc2e300335672
Top:    c956cfdc69fc0cc4c6d7a2a5780475601a0e3d87
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-31 02:10:09 +0100

Refresh of drc-fix-line-polygon-and-polyg

---

diff --git a/src/find.c b/src/find.c
index 5d2e92c..0a9de4c 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2684,7 +2684,7 @@ IsLineInPolygon (LineTypePtr Line, PolygonTypePtr Polygon)
       && Box->Y1 <= Polygon->Clipped->contours->ymax + Bloat
       && Box->Y2 >= Polygon->Clipped->contours->ymin - Bloat)
     {
-      if (!(lp = LinePoly (Line, Line->Thickness + Bloat)))
+      if (!(lp = LinePoly (Line, Line->Thickness + 2 * Bloat)))
         return FALSE;           /* error */
       return isects (lp, Polygon, true);
     }
@@ -2743,7 +2743,7 @@ IsPolygonInPolygon (PolygonTypePtr P1, PolygonTypePtr P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = 0; /* Bloat is added by IsLineInPolygon */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
